server/komodo-rest/src/main/java/org/komodo/servicecatalog/TeiidOpenShiftClient.java

```diff
@@ -1021,27 +1118,74 @@ public BuildStatus deleteVirtualization(String vdbName) {
         KubernetesClient kubernetesClient = new DefaultKubernetesClient(config);
         final OpenShiftClient client = kubernetesClient.adapt(OpenShiftClient.class);
         try {
-            KLog.getLogger().info("Deleting the " + vdbName + "that is deployed as Service");
+            info("Deleting the " + vdbName + "that is deployed as Service");
             if (runningBuild != null) {
-                client.builds().inNamespace(runningBuild.namespace).withName(runningBuild.buildName).delete();
+                client.builds().inNamespace(runningBuild.namespace()).withName(runningBuild.buildName()).delete();
             } else {
                 runningBuild = getVirtualizationStatus(vdbName);
             }
             client.buildConfigs().inNamespace(namespace).withName(getBuildConfigName(vdbName)).delete();
             client.deploymentConfigs().inNamespace(namespace).withLabel("application", vdbName).delete();
-            //client.routes().inNamespace(namespace).withName(vdbName+"-jdbc").delete();
-            client.routes().inNamespace(namespace).withName(vdbName+"-odata").delete();
-            client.services().inNamespace(namespace).withName(vdbName+"-jdbc").delete();
-            client.services().inNamespace(namespace).withName(vdbName+"-odata").delete();
-            client.services().inNamespace(namespace).withName(vdbName+"-odbc").delete();
+            //client.routes().inNamespace(namespace).withName(vdbName+HYPHEN + RouteType.JDBC.id()).delete();
+            client.routes().inNamespace(namespace).withName(vdbName+HYPHEN + ProtocolType.ODATA.id()).delete();
+            client.services().inNamespace(namespace).withName(vdbName+HYPHEN + ProtocolType.JDBC.id()).delete();
+            client.services().inNamespace(namespace).withName(vdbName+HYPHEN + ProtocolType.ODATA.id()).delete();
+            client.services().inNamespace(namespace).withName(vdbName+HYPHEN + ProtocolType.ODBC.id()).delete();
             client.imageStreams().inNamespace(namespace).withName(vdbName).delete();
         } finally {
             kubernetesClient.close();
         }
-        runningBuild.statusMessage = "deleted";
+        runningBuild.setStatusMessage("deleted");
         return runningBuild;
     }
 
+    private RouteStatus getRoute(String vdbName, ProtocolType protocolType) {
```

Comment:
Maybe it would have been better to just return the coordinates of service or route and let the consuming utility take care of getting to the actual hosts and ports. But I guess that's OK